napari/layers/vectors/vectors.py

```diff
@@ -242,9 +242,8 @@ def __init__(
 
         # Data containing vectors in the currently viewed slice
         self._view_data = np.empty((0, 2, 2))
-        self._displayed_stored = []
-        self._view_indices = []
-        self._view_alphas = []
+        self._view_indices: List[int] = []
+        self._view_alphas: List[float] = []
```

Comment:
Technically this is

```suggestion
        self._view_alphas: Union[List[float], float] = []
```

because `_view_alphas` can be set to `1.0` in `_set_view_slice` either directly or from `_slice_data`.

Alternatively we could consider typing this as `np.ndarray` because this is only used to modify the face colors and only needs to be appropriately broadcastable.

Reply:
Weird, don't know why this wasn't caught by mypy. I've updated to `Union[float, np.ndarray]`, which is what `_slice_data()` returns, and updated the initial assigment to be `= 1` as a reasonable default initial value.